prow/scheduler/reconciler_test.go

```diff
@@ -169,7 +171,11 @@ func TestReconcile(t *testing.T) {
 			}
 			pjClient := builder.Build()
 
-			r := scheduler.NewReconciler(pjClient, &fakeStrategy{cluster: tc.cluster, err: tc.schedulingError})
+			r := scheduler.NewReconciler(pjClient,
+				func() *config.Config { return nil },
+				func(_ *config.Config) strategy.Interface {
+					return &fakeStrategy{cluster: tc.cluster, err: tc.schedulingError}
+				})
```

Comment:
If my understanding is correct, `*config.Config` is supposed to be nil in case of error only (see [here](https://github.com/kubernetes/test-infra/blob/23b3708d798e15de739648fc2231f9d104ffab60/prow/config/config.go#L1648-L1675)). The rest of the codebase doesn't check for `nil` but I have no problem changing it if you spot any flaw.

> To improve test fidelity, could you provide a config getter that returns a config with the strategy, and use strategy.Get()?

I'm not so sure about what you mean here, would you mind providing an example?

Reply:
You're right, I see that other uses of config.Getter check neither the getter nor the return values for nil, so that is consistent.

My concern about fidelity in this test case is there is no check that the config getter is actually used. My suggestion was to couple them in a way that ensured it is used. It looks like the tests added below do verify that, so this is probably okay.